core/src/main/scala/chisel3/experimental/hierarchy/core/Instance.scala

```diff
@@ -93,6 +93,10 @@ object Instance extends SourceInfoDoc {
       case Clone(x: IsClone[_] with BaseModule) => x.toAbsoluteTarget
       case _ => throw new InternalErrorException("Match error: i.underlying=${i.underlying}")
     }
+
+    def suggestName(name: String): Unit = {
+      i.getInnerDataContext.get.asInstanceOf[ModuleClone[T]].suggestName(name)
```

Comment:
We need to add a test for what happens if you do `Module(new MyModule).toInstance.suggestName(...)`